lib/rules/no-uninstalled-addons.ts

```diff
@@ -0,0 +1,185 @@
+    }
+
+    type GetPackageJson = (path: string) => Record<string, any>
+
+    const getPackageJson: GetPackageJson = (path) => {
+      const packageJson = {
+        devDependencies: {},
+        dependencies: {},
+      }
+      try {
+        const file = readFileSync(path, 'utf8')
+        const parsedFile = JSON.parse(file)
+        packageJson.dependencies = parsedFile.dependencies || {}
+        packageJson.devDependencies = parsedFile.devDependencies || {}
+      } catch (e) {
+        console.error(
+          'Could not fetch package.json - it is probably not in the same directory as the .storybook folder'
+        )
+      }
+
+      return packageJson
+    }
+
+    //----------------------------------------------------------------------
+    // Public
+    //----------------------------------------------------------------------
+
+    return {
+      AssignmentExpression: function (node) {
+        // when this is running for .storybook/main.js, we get the path to the folder which contains the package.json of the
+        // project. This will be handy for monorepos that may be running ESLint in a node process in another folder.
+        const projectRoot = context.getPhysicalFilename
+          ? resolve(context.getPhysicalFilename(), '../../')
+          : './'
+
+        try {
+        } catch (e) {}
+        const packageJsonObject = getPackageJson(`${projectRoot}/package.json`)
+        const depsAndDevDeps = mergeDepsWithDevDeps(packageJsonObject)
+
+        if (isObjectExpression(node.right)) {
+          const addonsProp = node.right.properties.find(
+            (prop) => isProperty(prop) && isIdentifier(prop.key) && prop.key.name === 'addons'
+          ) as Property | undefined
+
+          if (addonsProp) {
+            if (isArrayExpression(addonsProp.value)) {
+              // extract all nodes taht are a string inside the addons array
+              const listOfAddonsInString: string[] = addonsProp.value.elements
+                .map((elem) => (isLiteral(elem) ? elem.value : undefined))
+                .filter((elem) => !!elem) as string[]
+
+              // extract all nodes that are an object inside the addons array
+              const listOfAddonsInObj = addonsProp.value.elements
+                .map((elem) => (isObjectExpression(elem) ? elem : { properties: [] }))
+                .map((elem) => {
+                  const property: Property = elem.properties.find(
+                    (prop) => isProperty(prop) && isIdentifier(prop.key) && prop.key.name === 'name'
+                  ) as Property
+                  return isLiteral(property?.value) ? property.value.value : ''
+                })
+                .filter((elem) => !!elem) as string[]
+
+              const listOfAddons = [...listOfAddonsInString, ...listOfAddonsInObj]
+
+              const result = areThereAddonsNotInstalled(listOfAddons, depsAndDevDeps)
+
+              result
+                ? context.report({
+                    node,
```

Comment:
This node refers to the assignment expression, which will highlight the entire main.js config as wrong.

Ideally we should highlight only the offending lines, to make it as clear as possible for the users where the error is at. Here are a couple suggestions:

1 - Use addonsProp instead of node. This is probably good enough
2 - Keep the reference of string literals, then loop through the wrong ones and use them instead of node. This way you will highlight only the bits that are truly wrong, and you can call report multiple times, showing all wrong entries rather than just the first one.